packages/aws-cdk/lib/api/util/cloudformation/stack-activity-monitor.ts

```diff
@@ -420,6 +420,10 @@ abstract class ActivityPrinterBase implements IActivityPrinter {
       delete this.resourcesInProgress[activity.event.LogicalResourceId];
     }
 
+    if (status.endsWith('_COMPLETE_CLEANUP_IN_PROGRESS')) {
+      this.resourcesDone++;
+    }
```

Comment:
This catches both `ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS` and `UPDATE_COMPLETE_CLEANUP_IN_PROGRESS` statuses. 

These statuses are expected to be treated as a `resourceDone` tally in order for us to meet the `this.resourcesTotal`. 

I'm not entirely sure why[ `this.resourcesTotal`](https://github.com/aws/aws-cdk/blob/16b8a4e24d13948c9a7092df183693ddd2d4be0b/packages/aws-cdk/lib/api/util/cloudformation/stack-activity-monitor.ts#L387-L389) is accounting for `_COMPLETE_CLEANUP_IN_PROGRESS` status. The value comes from the `changes` property from calling the SDK `describeStacks()` API. Is it a count of `COMPLETE` occurrences to expect?

Reply:
This comment is fantastic and should go into the code! 😄

Reply:
Also, you have my blessing to reduce `this.resourcesTotal` by one... but then you'd have to figure out whether or not this is an update to an existing stack.

Will stacks that don't have anything to clean up also go through `_CLEANUP_IN_PROGRESS` ?

Reply:
> This comment is fantastic and should go into the code! 

👍🏻 I'll get that comment added to the code in another PR. 

> Will stacks that don't have anything to clean up also go through _CLEANUP_IN_PROGRESS ?

I believe `_CLEANUP_IN_PROGRESS` is only called on updates or rollback failures. 

[CFN definition](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-console-view-stack-data-resources.html): "Ongoing removal of old/new resources for one or more stacks after a successful/failed stack update."

I could be wrong but it seems to me that the `resourcesTotal` count ultimately comes from the estimated number of `*_COMPLETE*` events to be executed during the CloudFormation process. Which is why I'm hesitant to reduce `resourcesTotal` by 1. `resourcesTotal` should really be renamed to `changesTotal` because the value comes from a property called `changes` returned from our SDK API and does not seem to be an accurate count of the total "resources" cloudformation will create/modify.